e2e/testcases/multi_sync_test.go

```diff
@@ -399,10 +399,21 @@ func TestConflictingDefinitions_RootToRoot(t *testing.T) {
 	nt.RootRepos[rootSync2].CommitAndPush("add conflicting pod owner role")
 
 	nt.T.Logf("The admission webhook should deny the update request in Root %s", rootSync2)
-	nt.WaitForRootSyncSyncError(rootSync2, applier.ApplierErrorCode, "denied the request", false)
-	nt.T.Logf("Root %s should also get surfaced with the conflict error", configsync.RootSyncName)
-	// Ignore the Syncing condition because the sync status is updated by another reconciler, which doesn't update the SyncingCondition.
-	nt.WaitForRootSyncSyncError(configsync.RootSyncName, status.ManagementConflictErrorCode, "declared in another repository", true)
+	nt.WaitForRootSyncSyncError(rootSync2, applier.ApplierErrorCode, "denied the request")
+
+	// The first RootSync to apply the object will apply its manager annotation,
+	// which causes the webhook to reject attempts to update the object by other
+	// RootSyncs, the manager annotation won't change, so the first RootSync
+	// won't get any management conflict errors.
+	// The first RootSync would only get an error if the webhook was disabled.
+
+	// TODO: Should adoption fights happen with the webhook enabled?
+	// The remediator doesn't return a KptManagementConflictError, so the
+	// following error validation will never succeed.
+
+	// nt.T.Logf("Root %s should also get surfaced with the conflict error", configsync.RootSyncName)
+	// nt.WaitForRootSyncSyncError(configsync.RootSyncName, status.ManagementConflictErrorCode, "declared in another repository")
```

Comment:
AFAIK, the `declared in another repository` text only comes from the `KptManagementConflictError`, which only comes from the Applier, not the Remediator. If the manager is correct, the error won't be returned unless the object is adopted. But it can't be adopted with the webhook enabled, because the webhook rejects it, causing the `KptManagementConflictError` in the second RootSync, but not the first.

So the previous test validation seems impossible. Either the test is wrong and only worked because the status updates were flakey, OR the behavior is wrong and we need to change either the webhook or the remediator to make the test behavior reality.